developers/contracts/xlink-staking.md

```diff
@@ -0,0 +1,458 @@
+
+### `total-staked`
+
+| Data | Type             |
+| ---- | ---------------- |
+| Map  | `principal uint` |
+
+Tracks the total value staked for each token contract (`principal`).
+
+### `total-shares`
+
+| Data | Type             |
+| ---- | ---------------- |
+| Map  | `principal uint` |
+
+Tracks the total shares issued for each staking token (`principal`).
+
+### `validators-registry`
+
+| Data | Type                                                |
+| ---- | --------------------------------------------------- |
+| Map  | `principal { token: principal, pubkey: (buff 33) }` |
+
+Maintains a mapping of the registered validators. Each one is identified by a Stacks `principal` and has associated `token` and public key used for message signature verification.
+
+### `required-validators`
+
+| Data     | Type   |
+| -------- | ------ |
+| Variable | `uint` |
+
+Indicates the minimum number of validators required to sign a message for it to be considered valid.
+
+### `accrued-rewards`
+
+| Data | Type                                             |
+| ---- | ------------------------------------------------ |
+| Map  | `principal { amount: uint, update-block: uint }` |
+
+Tracks the total rewards that have been collected and restaked for each token, identified by its contract `principal`. The `update-block` field records the [`stacks-block-height`](https://docs.stacks.co/reference/keywords#stacks-block-height) of the last update. This map plays a key role in the [`add-rewards`](#add-rewards) function mechanism.
+
+### `block-threshold`
+
+| Data     | Type   |
+| -------- | ------ |
+| Variable | `uint` |
+
+Specifies the number of Stacks blocks allowed as a delay for submitting an [`add-rewards`](#add-rewards) message.
+
+### `approved-updaters`
+
+| Data | Type             |
+| ---- | ---------------- |
+| Map  | `principal bool` |
+
+Maps updaters (`principal`) to their approval status (`bool`).
+
+### Relevant constants
+
+#### `ONE_8`
+
+| Type   | Value        |
+| ------ | ------------ |
+| `uint` | `u100000000` |
+
+Mathematical constant used to restrict the decimal precision to 8 digits.
+
+#### `MAX_REQUIRED_VALIDATORS`
+
+| Type   | Value |
+| ------ | ----- |
+| `uint` | `u20` |
+
+Defines the upper bound for the [`required-validators`](#required-validators) variable.
+
+## Contract calls
+
+- `<ft-trait>`: Interaction with any approved token to perform mint actions ([`add-rewards`](#add-rewards)) and transfers (mostly within [`stake`](#stake) and [`unstake`](#unstake), but also within the [`withdraw`](#withdraw) governance function). The `ft-trait` within the contract is the custom SIP-010 implementation for handling fixed notation, available at `'SP2XD7417HGPRTREMKF748VNEQPDRR0RMANB7X1NK.trait-sip-010`.
+
+- `'SP2XD7417HGPRTREMKF748VNEQPDRR0RMANB7X1NK.executor-dao`: This contract is exclusively called by the [`is-dao-or-extension`](#is-dao-or-extension) function for authorizing governance operations.
+
```

Comment:
Are there no connection to registry contract?

Reply:
Nope, there is no registry for this contract.